### PR TITLE
feat(cli): implement node setup command

### LINKED ISSUE #471: implement node setup command

### PR DESCRIPTION
**Pull request by leninmehedy**
## Description

This pull request changes the following:

- `node setup` command
- few new modules
- code cleanup

This PR is already very large. I'll start polishing separately since this has the first working copy of the `node setup` command.

### Related Issues

- Closes #471

### CODE REVIEW

File: fullstack-network-manager/src/commands/init.mjs [hunk 1]
```diff
@@ -1,37 +1,74 @@
...
+            return false
+        }
     }
 
     /**
      * Return Yargs command definition for 'init' command
      * @param initCmd an instance of InitCommand
      */
-    static getCommandDefinition(initCmd){
+    static getCommandDefinition(initCmd) {
```
Comment (jeromy-cannon) [inline fullstack-network-manager/src/commands/init.mjs:71]: should we be adding some type of linter or spotless check?

File: fullstack-network-manager/src/commands/node.mjs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+                            nodeCmd.stop(argv).then(r => {
+                                nodeCmd.logger.debug("==== Finished running `node stop`====")
+
+                                if (!r) process.exit(1)
+                            })
+
+                        }
+                    })
+                    .demand(1, 'Select a node command')
```
Comment (jeromy-cannon) [inline fullstack-network-manager/src/commands/node.mjs:123]: FYI, I noticed that `.demand()` is showing as deprecated when I stepped into it to get the documentation on it.

File: fullstack-network-manager/src/core/constants.mjs [hunk 3]
```diff
@@ -4,6 +4,8 @@ import {fileURLToPath} from "url"
 // directory of this fle
 const CUR_FILE_DIR = dirname(fileURLToPath(import.meta.url))
 const USER = `${process.env.USER}`
+const FST_HOME_DIR = `${process.env.HOME}/.fsnetman`
```
Comment (jeromy-cannon) [inline fullstack-network-manager/src/core/constants.mjs:7]: do we want to give the user the ability to override the FST_HOME_DIR location if they want to?  Perhaps only set it if it isn't already set?

File: fullstack-network-manager/README.md [hunk 4]
```diff
@@ -49,6 +26,16 @@ Options:
   -v, --version  Show version number                                                                           [boolean]
 
 Select a command
-
 ```
 
+## Develop
```
[the remark this answers is not in the document]
Comment (leninmehedy) [reply fullstack-network-manager/README.md:31]: Thanks. Added some notes.

File: fullstack-network-manager/README.md [hunk 5]
```diff
@@ -49,6 +26,16 @@ Options:
...
+- In order to support ES6 modules with `jest`, set an env variable `export NODE_OPTIONS=--experimental-vm-modules >> ~/.zshrc`
+  - If you are using Intellij and would like to use debugger tools, you will need to enable `--experimental-vm-modules` for `Jest`.
+    - `Run->Edit Configurations->Edit Configuration Templates->Jest` and then set `--experimental-vm-modules` in `Node Options`.
+- Run `npm i` to install the required packages
+- Run `npm link` to install `fsnetman` as the CLI 
+  - Note: you need to do it once. If `fsnetman` already exists in your path, you will need to remove it first.
+- Run `npm test` or `npm run test` to run the unit tests
+- Run `npm run test-e2e` to run the long-running integration tests
+- Run `npm run fsnetman` to access the CLI as shown above
```
[the remark this answers is not in the document]
Comment (leninmehedy) [reply fullstack-network-manager/README.md:40]: Thanks. Added some notes.

File: fullstack-network-manager/src/core/constants.mjs (hunk 3 above)
Comment (leninmehedy) [reply fullstack-network-manager/src/core/constants.mjs:7]: This is not critical atm. Added an issue to handle it later.

File: fullstack-network-manager/src/commands/init.mjs (hunk 1 above)
Comment (leninmehedy) [reply fullstack-network-manager/src/commands/init.mjs:71]: Created a new issue

File: fullstack-network-manager/src/commands/node.mjs (hunk 2 above)
Comment (leninmehedy) [reply fullstack-network-manager/src/commands/node.mjs:167]: Thanks. fixed.

**Review by jeromy-cannon — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Pull request merged**